url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FisImporter.java

```diff
@@ -0,0 +1,626 @@
+                inputVariable.setName(Op.makeValidId(value));
+            } else if ("Enabled".equals(key)) {
+                inputVariable.setEnabled(Op.isEq(Op.toDouble(value), 1.0));
+            } else if ("Range".equals(key)) {
+                Op.Pair<Double, Double> minmax = extractRange(value);
+                inputVariable.setMinimum(minmax.first);
+                inputVariable.setMaximum(minmax.second);
+            } else if (key.startsWith("MF")) {
+                inputVariable.addTerm(prepareTerm(extractTerm(value), engine));
+            } else if ("NumMFs".equals(key)) {
+                //ignore
+            } else {
+                throw new RuntimeException(String.format(
+                        "[import error] token <%s> not recognized", key));
+            }
+        }
+    }
+
+    protected void importOutput(String section, Engine engine) throws Exception {
+        BufferedReader reader = new BufferedReader(new StringReader(section));
+        reader.readLine(); //ignore first line [InputX]
+
+        OutputVariable outputVariable = new OutputVariable();
+        engine.addOutputVariable(outputVariable);
+
+        String line;
+        while ((line = reader.readLine()) != null) {
+            List<String> keyValue = Op.split(line, "=");
+            if (keyValue.size() != 2) {
+                throw new RuntimeException(String.format(
+                        "[syntax error] expected a property of type "
+                        + "'key=value', but found <%s>", line));
+            }
+            String key = keyValue.get(0).trim();
+            String value = keyValue.get(1).trim();
+
+            if ("Name".equals(key)) {
+                outputVariable.setName(Op.makeValidId(value));
+            } else if ("Enabled".equals(key)) {
+                outputVariable.setEnabled(Op.isEq(Op.toDouble(value), 1.0));
+            } else if ("Range".equals(key)) {
+                Op.Pair<Double, Double> minmax = extractRange(value);
+                outputVariable.setMinimum(minmax.first);
+                outputVariable.setMaximum(minmax.second);
+            } else if (key.startsWith("MF")) {
+                outputVariable.addTerm(prepareTerm(extractTerm(value), engine));
+            } else if ("Default".equals(key)) {
+                outputVariable.setDefaultValue(Op.toDouble(value));
+            } else if ("LockValid".equals(key)) {
+                outputVariable.setLockValidOutput((int) Op.toDouble(value) == 1);
+            } else if ("LockRange".equals(key)) {
+                outputVariable.setLockOutputRange((int) Op.toDouble(value) == 1);
+            } else if ("NumMFs".equals(key)) {
+                //ignore
+            } else {
+                throw new RuntimeException(String.format(
+                        "[import error] token <%s> not recognized", key));
+            }
+        }
+    }
+
+    protected void importRules(String section, Engine engine) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.